### PR TITLE
openthread: samples: Added using UART HWFC by default.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Overlays setting UART hardware flow control by default
were added/extended.

KRKNWK-7370

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

**Review by MarekPorwisz — CHANGES_REQUESTED** [review]
what about coap client?

**Comment by kkasperczyk-no** [comment]
> what about coap client?

Right, I forgot that coap client is not always working as SED and may use UART too.

**Review by MarekPorwisz — APPROVED** [review]

**Comment by MarekPorwisz** [comment]
Don't we need to update some doc to inform user that HWFC needs to be enabled for Thread samples?

**Comment by kkasperczyk-no** [comment]
> Don't we need to update some doc to inform user that HWFC needs to be enabled for Thread samples?

@MarekPorwisz it wasn't considered when the task was planned, but actually I think it could be useful. On the other hand it requires also adding some kind of guide for user, how to configure HWFC support on PC and boards' JLink. Do you think we could merge this as it is and create new task on documentation after consulting with tech writer or redefine it and merge it including complete documentation?

**Comment by kkasperczyk-no** [comment]
I placed mention in each Thread sample informing that by default HWFC mechanism is enabled.

File: samples/openthread/coap_server/README.rst [hunk 1]
```diff
@@ -49,6 +49,8 @@ LED 3:
 LED 4:
   Turned on and off by messages sent from the client nodes.
 
+Sample by default uses hardware flow control mechanism in serial communication.
```
Comment (MarekPorwisz) [inline samples/openthread/coap_server/README.rst:52]: Move this somewhere around where putty is mentioned.

File: samples/openthread/coap_client/README.rst [hunk 2]
```diff
@@ -57,6 +57,8 @@ Button 2:
 Button 4:
    Send a multicast pairing request to the ``/provisioning`` resource.
 
+Sample by default uses hardware flow control mechanism in serial communication.
```
Comment (MarekPorwisz) [inline samples/openthread/coap_client/README.rst:60]: Move this somewhere around where putty is mentioned i.e.  ~line 139

File: samples/openthread/cli/README.rst [hunk 3]
```diff
@@ -51,6 +51,7 @@ User interface
 All the interactions with the application are handled using serial communication.
 See `OpenThread CLI Reference`_ for the list of available serial commands.
 
+Sample by default uses hardware flow control mechanism in serial communication.
```
Comment (MarekPorwisz) [inline samples/openthread/cli/README.rst:54]: Move this somewhere around where putty is mentioned. (Probably multiple places in this sample)

**Review by MarekPorwisz — CHANGES_REQUESTED** [review]

**Review by greg-fer — APPROVED** [review]
Approving, but note there is a follow-up task for after 1.4 release: https://projecttools.nordicsemi.no/jira/browse/KRKNWK-7447

**Review by MarekPorwisz — APPROVED** [review]

**Comment by kkasperczyk-no** [comment]
@carlescufi it's approved and all green, could you merge it, as I'm waiting for this change?

**Review by rlubos — APPROVED** [review]

**Pull request merged**